frontends/web/src/contexts/DarkmodeProvider.tsx

```diff
@@ -43,26 +44,37 @@ export const DarkModeProvider = ({ children }: TProps) => {
     getConfig()
       .then(config => {
         // use config if it exists
-        if ('darkmode' in config.frontend) {
+        if (!!config.frontend && 'darkmode' in config.frontend) {
           setIsDarkMode(config.frontend.darkmode);
           return;
         }
         // else use mode from OS
-        setIsDarkMode(osPrefersDarkmode);
-
+        if (runningInAndroid()) {
+          setIsDarkMode(androidPrefersDarkMode);
+        } else {
+          detectDarkTheme().then(isDark => {
+            setIsDarkMode(isDark);
+          });
+        }
       })
       .catch(console.error);
-  }, [osPrefersDarkmode]);
+  }, [androidPrefersDarkMode]);
 
   useEffect(() => {
     setAppTheme();
   }, [isDarkMode, setAppTheme]);
 
   const toggleDarkmode = (darkmode: boolean) => {
-    setIsDarkMode(!isDarkMode);
+    setIsDarkMode(darkmode);
     getConfig()
-      .then(config => {
-        if (osPrefersDarkmode === darkmode) {
+      .then(async config => {
+        let preferredDarkMode;
+        if (runningInAndroid()) {
+          preferredDarkMode = androidPrefersDarkMode;
+        } else {
+          preferredDarkMode = await detectDarkTheme();
+        }
+        if (preferredDarkMode === darkmode) {
```

Comment:
Not necessarily wrong to mix `await` and `.then()` but generally we try to stick to one convention, especially in a single function. 🙂